### DISCUSSION TITLE
Copilot 1.73.3 is using an old vunerable version of Nodejs 16.15.0

### DISCUSSION THREAD

**Discussion by fredrik666-6** [community/community#48149] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

My security department blocks Copilot due to the version of Nodejs (16.15.0) has known critical vulnerabilities.

Is there a way to change override the Nodejs version used by Copilot for Visual Studio? 16.17.1 would be a good replacement from a security perspective.

**Comment by Akash1134** [comment, marked as answer]
Hi @fredrik666-6 Thanks for reporting this. Are you using the latest version of GitHub Copilot? The latest is 1.73.8685. How did you check the [Nodejs version ](https://nodejs.org/en/) and is it the latest you can install?

**Reply by fredrik666-6** [reply]
Hi @Akash1134 

Thanks for your response!

The version i've downloaded is for standard Visual Studio (not VSCode) and i did it through the extension manager within the VS (Visual Studio Marketplace) and the latest version there is 1.73.3.0.
https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs

The security issue is detected by the security departments logs in FortiEDR (or some other Forti product). 

The only Nodejs version i can see i have installed is 16.17.1 but given this issue i assumed Copilot uses some internally bundled version?

brgds,
fredrik

**Reply by fredrik666-6** [reply]
I've downloaded the latest Nodejs 19, reinstalled Copilot but no change.

**Reply by Akash1134** [reply]
Hi @fredrik666-6 , Visual Studio Copilot runs an agent. The agent is basically a binary packed with everything Copilot needs. This includes node. The node version we use for packaging right now is 16.19.1.

**Reply by fredrik666-6** [reply]
Hi @Akash1134, 
Ok, that explains why it does not care about my downloaded version.
Node version 16.19.1, does that also apply to the GitHub.copilotVS version which is only available in version 1.73.3.0.?
brgds,
Fredrik
